src/averageAges.js

```diff
@@ -37,9 +45,23 @@
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let womens = people.filter(person => person.sex === 'f');
+
+  if (withChildren === true) {
+    womens = womens.filter(woman => people.some(person =>
+      person.mother === woman.name));
+  }
+
+  const totalAge = womens.reduce((sum, person) => sum
+    + (person.died - person.born), 0);
+
+  const averageAge = parseFloat((totalAge / womens.length).toFixed(2));
+
+  return averageAge;
```

Comment:
Similar to the previous function, consider renaming `calculateWomenAverageAge` for clarity. The average age calculation is correct, but the same suggestion to simplify applies.